samples/snippets/src/main/java/com/example/spanner/HubbleTransactionsCodeLab.java

```diff
@@ -87,55 +87,102 @@ public void writeWorkItems(
-    //                      Random random = new Random();
-    //                      ResultSet resultSet = transaction.executeQuery(Statement.of(stmt));
-    //                      List<List<Object>> rows = resultSet.getRows();
-    //                      resultSet.close();
-    //                      if (rows.size() == 0) return false;
-    //                      int randIdx = random.nextInt(resultSet.size());
-    //                      String id = rows.get(randIdx).get(0);
-    //                      System.out.printf("Doing work: %s\n", id);
-    //                      transaction.buffer(
-    //                              Mutation.newUpdateBuilder("WorkList")
-    //                                      .set("id").to(id)
-    //                                      .set("is_done").to(doneValue)
-    //                                      .build());
-    //                      return true;
-    //                    });
-    //    return didWork;
-    return true;
+    List<String> idList = new ArrayList<>();
+
+    try (ReadOnlyTransaction transaction = dbClient.readOnlyTransaction()) {
+      ResultSet queryResultSet = transaction.executeQuery(Statement.of(stmt));
+      while (queryResultSet.next()) {
+        idList.add(queryResultSet.getString("id"));
+      }
+      queryResultSet.close();
+    }
+    if (idList.size() == 0) return false;
+
+    return Boolean.TRUE.equals(
+        dbClient
+            .readWriteTransaction()
+            .run(
+                transaction -> {
+                  Random random = new Random();
+                  int randIdx = random.nextInt(idList.size());
+                  String id = idList.get(randIdx);
+                  Struct idRow =
+                      transaction.readRow(
+                          "WorkList",
+                          Key.of(id), // Read single row in a table.
+                          Arrays.asList("id", "is_done", "timestamp"));
+
+                  if (idRow == null || idRow.getBoolean("is_done")) return false;
+
+                  System.out.printf("Doing work: %s\n", id);
+                  transaction.buffer(
+                      Mutation.newUpdateBuilder("WorkList")
+                          .set("id")
+                          .to(idRow.getValue("id"))
+                          .set("is_done")
+                          .to(doneValue)
+                          .set("timestamp")
+                          .to(Value.COMMIT_TIMESTAMP)
+                          .build());
+                  return true;
+                }));
   }
 
-  public void doWorkSingleTransactionSerial(DatabaseClient dbClient, boolean doneValue) {
+  private void doWorkSingleTransactionSerial(
+      DatabaseClient dbClient, boolean doneValue, boolean isLocking) {
     boolean workRemaining = true;
     int workDone = 0;
     while (workRemaining) {
-      workRemaining = doWorkSingleTransaction(dbClient, doneValue);
-      if (workDone++ % 1000 == 0) {
```

Comment:
As this count will be used by individual thread so it will not reach 1000, it may be different for different thread. Hence I remove the print statement. If we really need to print something like it we may need add a logic or if printing individual thread is fine then I can add something here it self.

Reply:
This was a debugging statement, totally fine to remove :)